internal/cpugroup/cpugroup.go

```diff
@@ -0,0 +1,76 @@
+package cpugroup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const NullGroupID = "00000000-0000-0000-0000-000000000000"
+
+// ErrHVStatusInvalidCPUGroupState corresponds to the internal error code for HV_STATUS_INVALID_CPU_GROUP_STATE
+var ErrHVStatusInvalidCPUGroupState = errors.New("The hypervisor could not perform the operation because the CPU group is entering or in an invalid state.")
+
+// DeleteCPUGroup deletes the cpugroup from the host
+func DeleteCPUGroup(ctx context.Context, id string) error {
+	operation := hcsschema.DeleteGroup
+	details := hcsschema.DeleteGroupOperation{
+		GroupId: id,
+	}
+
+	return modifyCPUGroupRequest(ctx, operation, details)
+}
+
+// modifyCPUGroupRequest is a helper function for making modify calls to a cpugroup
+func modifyCPUGroupRequest(ctx context.Context, operation hcsschema.CPUGroupOperation, details interface{}) error {
+	req := hcsschema.ModificationRequest{
+		PropertyType: hcsschema.PTCPUGroup,
+		Settings: &hcsschema.HostProcessorModificationRequest{
+			Operation:        operation,
+			OperationDetails: details,
+		},
+	}
+
+	return hcs.ModifyServiceSettings(ctx, req)
+}
+
+// CreateNewCPUGroupWithID creates a new cpugroup on the host with a prespecified id
+func CreateNewCPUGroupWithID(ctx context.Context, id string, logicalProcessors []uint32) error {
+	operation := hcsschema.CreateGroup
+	details := &hcsschema.CreateGroupOperation{
+		GroupId:               strings.ToLower(id),
+		LogicalProcessors:     logicalProcessors,
+		LogicalProcessorCount: uint32(len(logicalProcessors)),
+	}
+	if err := modifyCPUGroupRequest(ctx, operation, details); err != nil {
+		return fmt.Errorf("failed to make cpugroups CreateGroup request with details %v with: %s", details, err)
+	}
+	return nil
+}
+
+// getCPUGroupConfig finds the cpugroup config information for group with `id`
+func getCPUGroupConfig(ctx context.Context, id string) (*hcsschema.CpuGroupConfig, error) {
```

Comment:
Is there no use for this outside of this package besides for tests?

Reply:
Correct. There used to be more issues when we planned on managing these ourselves, but since we've decided that's the orchestrator's responsibility, we don't really care about what the cpugroup settings are.

Reply:
Thanks for clarifying :)